client/app/scripts/liveblog-edit/freetype.service.js

```diff
@@ -0,0 +1,537 @@
+                if (parts[i].indexOf('[') !== -1) {
+                    vector = parts[i].match(/([^\]]*)\[([^\]]*)]/);
+                    if (parseInt(vector[2], 10) == vector[2]) {
+                        vector[2] = parseInt(vector[2], 10);
+                    }
+                    part = {};
+                    // if the array is already set, just use that.
+                    if (angular.isArray(variable[vector[1]])){
+                        // is the array is set but not that index, make it.
+                        if (angular.isDefined(variable[vector[1]][vector[2]])) {
+                            variable = variable[vector[1]][vector[2]]
+                        } else {
+                            variable[vector[1]][vector[2]] = part
+                            variable = part;
+                        }
+                    } else {
+                        variable[vector[1]] = [];
+                        variable[vector[1]][vector[2]] = part
+                        variable = part;
+                    }
+                } else {
+                    // if the object is already set, just use that.
+                    if (angular.isDefined(variable[parts[i]])) {
+                        variable = variable[parts[i]];
+                        if (i === parts.length - 1) {
+                            return variable;
+                        }
+                    } else if (i === parts.length - 1){
+                        variable[parts[i]] = value || '';
+                    } else {
+                        variable[parts[i]] = {};
+                        variable = variable[parts[i]];
+                    }
+                }
+            }
+        }
+    }
+    /**
+    * Create a list of paths from `obj` parameter and return it in the
+    * `ret` parameter
+    */
+    function obj2path(ret, obj, path) {
+        ret = ret || {};
+        var gotopath, path;
+        angular.forEach(obj, function(value, key) {
+            if (angular.isArray(value)) {
+                for (var i = 0; i< value.length; i++) {
+                    gotopath = path? path + '.' + key + '[' + i + ']' : key + '[' + i + ']';
+                    obj2path(ret, value[i], gotopath);
+                }
+            } else if (angular.isObject(value)) {
+                gotopath = path? path + '.' + key : key;
+                obj2path(ret, value, gotopath);
+            } else {
+                gotopath = path? path + '.' + key : key;
+                ret[gotopath] = value;
+            }
+        });
+    }
+    /**
+     * Inject `class` in the class attributes string.
+     */
+    function injectClass(attr, cls) {
+        if (attr.indexOf('class') > -1) {
+            return attr.replace(/class\w*=\w*("|')?([^\"\']+)("|')/, 'class="' + cls + ' $2"');
+        } else {
+            attr += 'class="' + cls + '" ';
+            return attr;
+        }
+    }
+
+    angular.module('liveblog.freetype', ['liveblog.edit'])
```

Comment:
If freetype is a separate module, it should be put outside of `liveblog-edit` in a `liveblog-syndication` folder